extensions-core/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleDataSegmentKiller.java

```diff
@@ -70,12 +70,12 @@ public void kill(DataSegment segment) throws SegmentLoadingException
       // anymore, but we still delete them if exists.
       deleteIfPresent(bucket, descriptorPath);
     }
-    catch (IOException e) {
+    catch (StorageException e) {
```

Comment:
Why this change?

Reply:
`deleteIfPresent` no longer throws IOException

Reply:
the underlying lib could still throw an IOException though?

Reply:
Nope, it can't -- else it has to be caught or declared thrown since it's a checked exception.

Reply:
Currently wIth retry, only 2 exception kinds are possible: a) Exception raised by the retried call (only StorageException in this case), and b) RuntimeException. https://github.com/apache/druid/blob/bf39c71d2a51a9bd48ce8971060442e29a721b2f/processing/src/main/java/org/apache/druid/java/util/common/RetryUtils.java#L140
Do we still want to include IOException handling?

Reply:
Hmm. probably ok to leave.